File: benchmarks/cpu/divsum/divsum.c
#include <stdio.h>

long divsum(int n) {
    long total = 0;
    for (int i = 1; i <= n; i++) {
        long s = 0;
        for (int j = 1; j <= i; j++) {
            if (i % j == 0) s += j;
        }
        total += s;
    }
    return total;
}

int main(void) {
    printf("%ld\n", divsum(10000));
    return 0;
}

File: benchmarks/cpu/divsum/divsum.rs
fn divsum(n: i64) -> i64 {
    let mut total: i64 = 0;
    for i in 1..=n {
        let mut s: i64 = 0;
        for j in 1..=i {
            if i % j == 0 { s += j; }
        }
        total += s;
    }
    total
}

fn main() {
    println!("{}", divsum(10000));
}
